app/models/mixins/container_resource_parent_mixin.rb

```diff
@@ -0,0 +1,109 @@
+# ContainerResourceParentMixin provides capabilities to container
+# resources that provide hierarchical management to other container
+# resources.
+module ContainerResourceParentMixin
+  extend ActiveSupport::Concern
+  require 'json'
+  def get_resource_by_name(name, kind, api_version = 'v1')
```

Comment:
some entities that are namespeced must also pass project name to this method.
I think it would be best to add it to the interface now